src/plugins/Network.js

```diff
@@ -463,13 +504,79 @@ module.exports = class Network extends OverlayPlugin {
             return;
         }
 
-        const options = this.select.getElementsByTagName('option');
+        const options = this.selectProfile.getElementsByTagName('option');
         for (let i = 0; i < options.length; i++) {
             const option = options[i];
             if (option.value === profile.name) {
                 option.selected = 'selected';
             }
         }
-        this.changeProfile();
+    }
+
+    /**
+     * Update mobile profile list UI according to the current active profile.
+     *
+     * @param {string} profile Profile name.
+     */
+    setActiveMobileProfile(profile) {
+        if (!profile) {
+            log.error('setActiveMobileProfile: Error : provided profile is empty');
+            return;
+        }
+        const mobileProfile = MOBILE_PROFILES.find((elem) => elem.name === profile);
+        if (!mobileProfile) {
+            log.error('setActiveMobileProfile: Error : unknown provided profile');
+            return;
+        }
+
+        const options = this.selectMobileProfile.getElementsByTagName('option');
+        for (let i = 0; i < options.length; i++) {
+            const option = options[i];
+            if (option.value === mobileProfile.name) {
+                option.selected = 'selected';
+                break;
+            }
+        }
+    }
```

Comment:
Using [find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) can remove a little bit of boilerplate:
```javascript
        const options = this.selectMobileProfile.getElementsByTagName('option');
        const option = options.find(option => option.value === mobileProfile.name);
        if (option) {
            option.selected = 'selected';
        } 
```

Reply:
it's an array of 7 elements

Reply:
Actually, given the existing code above which is using for loops, it's probably better to keep the for loop here too, for consistency.